src/common/buffer.cc

```diff
@@ -1315,6 +1315,12 @@ static simple_spinlock_t buffer_debug_lock = SIMPLE_SPINLOCK_INITIALIZER;
     return true;
   }
 
+  bool buffer::list::is_provided_buffer(const char *dst) const
+  {
+    assert(!_buffers.empty());
```

Comment:
Any particular reason for the assert?  Can't you just return false if the bl is empty?